### PR TITLE
Add slim quantization module

### PR DESCRIPTION
**Pull request by baiyfbupt**

### CODE REVIEW

File: deploy/slim/quantization/quant.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        'fetch_name_list':eval_fetch_name_list,\
+        'fetch_varname_list':eval_fetch_varname_list}
+
+    if train_alg_type == 'det':
+        program.train_eval_det_run(config, exe, train_info_dict, eval_info_dict)
+    else:
+        program.train_eval_rec_run(config, exe, train_info_dict, eval_info_dict)
+
+
+def test_reader():
```
Comment (LDOUBLEV) [inline deploy/slim/quantization/quant.py:182]: test_reader可以删了

File: ppocr/modeling/architectures/det_model.py [hunk 2]
```diff
@@ -67,6 +67,7 @@ def create_feed(self, mode):
 
         image = fluid.layers.data(
             name='image', shape=image_shape, dtype='float32')
+        image.stop_gradient = False
```
Comment (LDOUBLEV) [inline ppocr/modeling/architectures/det_model.py:70]: 静态图的变量默认的Stop gradient 就是False 了吧

File: deploy/slim/quantization/quant.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+    train_reader = reader_main(config=config, mode="train")
+    train_loader.set_sample_list_generator(train_reader, places=place)
+
+    eval_reader = reader_main(config=config, mode="eval")
+
+    exe = fluid.Executor(place)
+    exe.run(startup_program)
+
+    def pact(x, name=None):
```
Comment (LDOUBLEV) [inline deploy/slim/quantization/quant.py:80]: 函数放在外面，加上函数注释

File: ppocr/modeling/heads/rec_ctc_head.py [hunk 4]
```diff
@@ -35,12 +35,13 @@ def __init__(self, params):
         self.fc_decay = params.get("fc_decay", 0.0004)
 
     def __call__(self, inputs, labels=None, mode=None):
-        encoder_features = self.encoder(inputs)
-        if self.encoder_type != "reshape":
-            encoder_features = fluid.layers.concat(encoder_features, axis=1)
-        name = "ctc_fc"
-        para_attr, bias_attr = get_para_bias_attr(
-            l2_decay=self.fc_decay, k=encoder_features.shape[1], name=name)
+        with fluid.scope_guard("skip_quant"):
```
Comment (LDOUBLEV) [inline ppocr/modeling/heads/rec_ctc_head.py:38]: 加了scope后，变量名字会不会改变，是否影响加载之前的模型

File: deploy/slim/quantization/quant.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        executor=executor,
+        for_test=False,
+        return_program=True)
+
+    # compile program for multi-devices
+    train_compile_program = program.create_multi_devices_program(
+        quant_train_program, train_opt_loss_name, for_quant=True)
+
+    # dump mode structure
+    if config['Global']['debug']:
```
Comment (LDOUBLEV) [inline deploy/slim/quantization/quant.py:155]: 默认的配置文件没有debug这个参数，加载原始的配置文件会报错，建议加个判断

File: deploy/slim/quantization/quant.py (hunk 1 above)
Comment (baiyfbupt) [reply deploy/slim/quantization/quant.py]: done

File: ppocr/modeling/architectures/det_model.py (hunk 2 above)
Comment (baiyfbupt) [reply ppocr/modeling/architectures/det_model.py:70]: image
默认是True, PACT量化因为底层实现原因要求image的stop_gradient属性为False, 所以才有这里的改动

File: ppocr/modeling/heads/rec_ctc_head.py (hunk 4 above)
Comment (baiyfbupt) [reply ppocr/modeling/heads/rec_ctc_head.py:38]: name_scope才会改变变量名，这里不会改变，不影响其他任务

File: deploy/slim/quantization/quant.py (hunk 3 above)
Comment (baiyfbupt) [reply deploy/slim/quantization/quant.py]: done, thanks~

File: deploy/slim/quantization/quant.py (hunk 5 above)
Comment (baiyfbupt) [reply deploy/slim/quantization/quant.py]: 这个是从tools/train.py copy的，在量化过程没有用到过，所以先移除了
